runtime/recipe/type-checker.js

```diff
@@ -194,11 +194,11 @@ export class TypeChecker {
     let resolvedRight = right.type.resolvedType();
     let [leftType, rightType] = Type.unwrapPair(resolvedLeft, resolvedRight);
 
-    // an unconstrained variable is compatible with a set.
-    if (leftType.isVariable && !(leftType.variable.canReadSubset) && !(leftType.variable.canWriteSuperset) && rightType.isSetView)
-      return true;
-    if (rightType.isVariable && !(rightType.variable.canReadSubset) && !(rightType.variable.canWriteSuperset) && leftType.isSetView)
-      return true;
+    // a variable is compatible with a set only if it is unconstrained.
+    if (leftType.isVariable && rightType.isSetView)
+      return !leftType.variable.canReadSubset && !leftType.variable.canWriteSuperset;
```

Comment:
Des Morgans this sucka

!(leftType.variable.canReadSubset || leftType.variable.canWriteSuperset)